### PR TITLE
Add new `RSpec/Rails/MinitestAssertions` cop

### PR DESCRIPTION
**Pull request by ydah**
Partially addresses #1485
______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

- [x] Added the new cop to `config/default.yml`.
- [x] The cop is configured as `Enabled: pending` in `config/default.yml`.
- [x] The cop is configured as `Enabled: true` in `.rubocop.yml`.
- [x] The cop documents examples of good and bad code.
- [x] The tests assert both that bad code is reported and that good code is not reported.
- [x] Set `VersionAdded` in `default/config.yml` to the next minor version.

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -573,6 +573,12 @@ RSpec/MessageSpies:
   VersionAdded: '1.9'
   Reference: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/MessageSpies
 
+RSpec/MinitestAssertions:
+  Description: Check if using Minitest matchers.
+  Enabled: pending
+  VersionAdded: '2.16'
```
Comment (pirj) [inline config/default.yml:579]: We're merging swiftly lately, but just in case - `<<next>>`?

File: lib/rubocop/cop/rspec/minitest_assertions.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      #   expect(a).not_to eq(b)
+      #
+      class MinitestAssertions < Base
+        extend AutoCorrector
+
+        MSG = 'Use `%<prefer>s` to matcher.'
+        RESTRICT_ON_SEND = %i[assert_equal refute_equal].freeze
+
+        # @!method arguments(node)
+        def_node_matcher :arguments, <<-PATTERN
```
Comment (pirj) [inline lib/rubocop/cop/rspec/minitest_assertions.rb:26]:
```suggestion
        def_node_matcher :minitest_assertion, <<-PATTERN
```

File: lib/rubocop/cop/rspec/minitest_assertions.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          (send nil? {:assert_equal :refute_equal} $_ $_ $_?)
+        PATTERN
+
+        def on_send(node)
+          arguments(node) do |exp, act, msg|
```
Comment (pirj) [inline lib/rubocop/cop/rspec/minitest_assertions.rb:31]:
```suggestion
          minitest_assertion(node) do |expected, actual, failure_message|
```

File: lib/rubocop/cop/rspec/minitest_assertions.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            prefer = replacement(node, exp, act, msg.first)
+            add_offense(node, message: message(prefer)) do |corrector|
+              corrector.replace(node, prefer)
+            end
+          end
+        end
+
+        private
+
+        def replacement(node, exp, act, msg = nil)
```
Comment (pirj) [inline lib/rubocop/cop/rspec/minitest_assertions.rb:41]:
```suggestion
        def replacement(node, expected, actual, failure_message)
```
I've removed the default value as we seem to always pass it

File: lib/rubocop/cop/rspec/minitest_assertions.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      #
+      #   # good
+      #   expect(a).to eq(b)
+      #   expect(a).to(eq(b), "must be equal")
+      #   expect(a).not_to eq(b)
+      #
+      class MinitestAssertions < Base
+        extend AutoCorrector
+
+        MSG = 'Use `%<prefer>s` to matcher.'
```
Comment (pirj) [inline lib/rubocop/cop/rspec/minitest_assertions.rb:22]:
```suggestion
        MSG = 'Use `%<prefer>s`.'
```

**Review by pirj — APPROVED** [review]
Looks great, thank you!
Really impressed how simple the autocorrection is implemented

I dared to change "resolves" to "partially addresses" as I had more than just `assert_equal` in mind:
`assert_empty` - `to be_empty`
`assert_in_delta`/`assert_in_epsilon` - `to be_within(...).of(...)`
and more ([Minitest](http://docs.seattlerb.org/minitest/Minitest/Assertions.html), [RSpec](https://relishapp.com/rspec/rspec-expectations/v/3-9/docs/built-in-matchers))

But this can be done in a follow-up, this cop is a great start as is.

File: config/default.yml (hunk 1 above)
Comment (ydah) [reply config/default.yml]: That sounds better. We updated to `<<next>>`.

**Comment by ydah** [comment]
I rebased this PR.

**Comment by pirj** [comment]
Thank you for updating to to `RSpec/Rails/` namespace, @ydah !

**Review by bquorning — APPROVED** [review]
Thank you @ydah 🙏🏼

**Pull request merged**